slidez-core/src/main/java/com/binarystudio/academy/slidez/link/LinkGenerationService.java

```diff
@@ -0,0 +1,131 @@
+package com.binarystudio.academy.slidez.link;
+
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import com.binarystudio.academy.slidez.link.dto.LinkDto;
+import com.binarystudio.academy.slidez.link.exceptions.IncorrectLeaseDurationException;
+import com.binarystudio.academy.slidez.link.model.Link;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class LinkGenerationService {
+
+	private static final int MAX_COUNT_AVAILABLE_LINKS = 100;
+
+	private static final int MIN_LEASE_DURATION = 1;
+
+	private static final int MAX_LEASE_DURATION = 180;
+
+	@Autowired
+	private LinkRepository linkRepository;
+
+	/**
+	 * [On startup, on corn job] Generate extra links, if there are less than 100
+	 * available
+	 */
+	public void generateExtraLinks() {
+
+		int countAvailableLinks = this.linkRepository.getCountAvailableLinks();
+		if (countAvailableLinks < this.MAX_COUNT_AVAILABLE_LINKS) {
+			int countLinkForGenerate = this.MAX_COUNT_AVAILABLE_LINKS - countAvailableLinks;
+			String lastLink = getLastLink();
+			for (int i = 0; i < countLinkForGenerate; i++) {
+				String newLink = generateLink(lastLink);
+				this.linkRepository.save(new Link(UUID.randomUUID(), null, newLink, null));
+				lastLink = newLink;
+			}
+		}
```

Comment:
if (countAvailableLinks **>=** this.MAX_COUNT_AVAILABLE_LINKS) {
return;
}

int countLinkForGenerate = this.MAX_COUNT_AVAILABLE_LINKS - countAvailableLinks;
			String lastLink = getLastLink();
			for (int i = 0; i < countLinkForGenerate; i++) {
				String newLink = generateLink(lastLink);
				this.linkRepository.save(new Link(UUID.randomUUID(), null, newLink, null));
				lastLink = newLink;
			}